blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -559,4 +630,53 @@ public Boolean apply(Row<Locator, Long> row) {
         return excessEnumMetrics;
     }
 
+    public <T extends Rollup> Points<T> transformEnumValueHashesToStrings(MetricData metricData, ColumnList<Long> enumvalues) {
+        Map<Long, String> hash2enumValues = getEnumValueFromHashes(enumvalues);
+        Points<T> pointsEnum = new Points<T>();
+
+        Map<Long, Points.Point<T>> pointsMap = metricData.getData().getPoints();
+
+        for (Long timestamp : pointsMap.keySet()) {
+            BluefloodEnumRollup enumRollup = (BluefloodEnumRollup)pointsMap.get(timestamp).getData();
+            for (Long hash : enumRollup.getHashedEnumValuesWithCounts().keySet()) {
+                String enumValueString = hash2enumValues.get(hash);
+                enumRollup.getStringEnumValuesWithCounts().put(enumValueString, enumRollup.getHashedEnumValuesWithCounts().get(hash));
+                pointsEnum.add(new Points.Point<T>(timestamp,(T)enumRollup));
+            }
+        }
+        return pointsEnum;
+    }
+
+    private Map<Long, String> getEnumValueFromHashes(ColumnList<Long> enumValues) {
```

Comment:
This method is useful and should be made public.  I am going to need it in my PR.  And what happened to the "getColumnsFromEnumCF" method?  I am going to need that also.  Or should I just add that on my own?